### PR TITLE
Adding zstd extraction

### PR DESCRIPTION
**Pull request by gdesmar**
Adding support for zstandard archive (CybercentreCanada/assemblyline/issues/104) with internal tar archive.
Bugfix with _submit_extracted with symlinks, plus the symlink resultsection not being added

### CODE REVIEW

**Review by cccs-kevin — APPROVED** [review]

File: extract/extract.py [hunk 1]
```diff
@@ -256,6 +244,27 @@ def execute(self, request: ServiceRequest):
             extracted, password_protected = self.extract_zip(request, request.file_path, request.file_type)
             summary_section_heuristic = 19
 
+        if request.file_type in ["archive/gzip", "archive/zstd"] and not password_protected and len(extracted) == 1:
```
Comment (kam193) [inline extract/extract.py:247]: FYI: As I understand the code, this case is a special handling for `.tar.*` files, but I think it's too restricted: just the [Python standard](https://docs.python.org/3/library/tarfile.html?highlight=tarfile#module-tarfile) library supports gzip, bzip2 and lzma. I'd suggest either extending the list for all cases from the standard library, or even make it general for _every_ archive containing just the tarfile.

**Review by kam193 — COMMENTED** [review]
It's really nice to see the zstd support :mechanical_arm:

File: extract/extract.py (hunk 1 above)
Comment (gdesmar) [reply extract/extract.py:247]: Just tested with a bzip2, and we're handling it without jumping over the middle tar. Definitely a nice optimization. I'll widen the second stage tar extraction to all files types handled in Extract. Thank you for the review!

**Review by kam193 — APPROVED** [review]
LGTM

**Review by cccs-kevin — APPROVED** [review]

**Pull request merged**